internal/xerrors/type.go

```diff
@@ -0,0 +1,28 @@
+package xerrors
+
+import "fmt"
+
+// Type reports which error type
+type Type uint8
+
+const (
+	TypeNoError = Type(iota)
```

Comment:
What about iota + 1 or add TypeUndefined as zero item?

for use explicit values only?